### PR TITLE
Add reports calendar view

### LINKED ISSUE #1723: Epic 1704: Add engagements calendar
Use a calendar widget to display the engagements.

Basic features of the calendar component:
- [x] Month, week, day views
- [x] prev/next navigation
- [x] select today
- [x] display all events with the same style and same details in all views
- [x] On click on a date, go to the day view of that date
- [x] On click on an event, go to event show page

Where to use the calendar component on?
- [x] On every page where there is a report list, add a Calendar tab: (Calendar | Summary | Table | Map)
- [x] Make sure they also contain future engagements

- [x] Add list view (see the list option here: https://fullcalendar.io/#demos)
- [x] Per view type, it should be possible to define a max number of items to display
- [x] Display the total number of items (or, maybe better, display a link with "x more...")
- [x] Display reports as being full-day event when no duration available

### PR DESCRIPTION
**Pull request by maradragan**
This adds a calendar view to all places where a collection of reports is displayed. From now on the default view is the calendar view.

Closes #1723 

### User changes
Users can now see reports displayed on a calendar on the following pages:

1. on the reports collection displayed after having clicked on a dashboard link
2. on the My Reports page
3. on an organization's detail page
4. on the daily rollup page
5. on a person's detail page
6. on a task's detail page
7. on the reports search results page

### Admin changes
In addition to the User changes above, Admins can now also see reports displayed on a calendar on the following pages:

1. on an authorization group's display page
2. on an insights page

### CODE REVIEW

**Comment by maradragan** [comment]
> Times in the calendar are shown as AM/PM (or even '12a' in month view for an engagement at midnight); should be 24h format.

Fixed it.

> With `engagementsIncludeTimeAndDuration: false` in `anet.yml`, I still see a time (and duration?) being displayed.

Fixed it.

> Month/Week/Day view do not scale to fit on one screen.

Fixed the month view. When we will make the calendar responsive the week and day view will be considered.

**Comment by gjvoosten** [comment]
> > Times in the calendar are shown as AM/PM (or even '12a' in month view for an engagement at midnight); should be 24h format.
> 
> Fixed it.

I would prefer if it used the formatting strings from the dictionary. This would seem to be possible by using the `@fullcalendar/moment` plugin: https://fullcalendar.io/docs/date-formatting#formatting-strings

File: client/src/components/ReportCollection.js [hunk 1]
```diff
@@ -212,6 +218,44 @@ export default class ReportCollection extends Component {
...
+        return {
+          title: who + "@" + where,
+          start: moment(r.engagementDate).format("YYYY-MM-DD HH:mm"),
+          end: moment(r.engagementDate)
+            .add(r.duration, "minutes")
+            .format("YYYY-MM-DD HH:mm"),
+          url: Report.pathFor(r),
+          classNames: ["event-" + r.state.toLowerCase()],
+          extendedProps: { ...r },
+          allDay: Settings.engagementsIncludeTimeAndDuration
```
Comment (gjvoosten) [inline client/src/components/ReportCollection.js]: `?:`-expression can be simplified to `!Settings.engagementsIncludeTimeAndDuration || r.duration !== null`

Comment (maradragan) [reply client/src/components/ReportCollection.js]: done

**Comment by maradragan** [comment]
> > > Times in the calendar are shown as AM/PM (or even '12a' in month view for an engagement at midnight); should be 24h format.
> > 
> > 
> > Fixed it.
> 
> I would prefer if it used the formatting strings from the dictionary. This would seem to be possible by using the `@fullcalendar/moment` plugin: https://fullcalendar.io/docs/date-formatting#formatting-strings and https://fullcalendar.io/docs/moment-plugins

I am making an issue for this, as I think it can be merged for now also as it is.

**Pull request merged**